diff --git a/plugins/carbon/carbon.c b/plugins/carbon/carbon.c
index 670b61c..c723120 100644
--- a/plugins/carbon/carbon.c
+++ b/plugins/carbon/carbon.c
@@ -259,7 +259,7 @@ static void carbon_push_stats(int retry_cycle, time_t now) {
 
 		int do_avg_push;
 
-		wok = carbon_write(fd, "%s%s.%s.requests %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, (unsigned long long) uwsgi.workers[0].requests, (unsigned long long) now);
+		wok = carbon_write(fd, "%s.%s%s.requests %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, (unsigned long long) uwsgi.workers[0].requests, (unsigned long long) now);
 		if (!wok) goto clear;
 
 		for(i=1;i<=uwsgi.numproc;i++) {
@@ -298,14 +298,14 @@ static void carbon_push_stats(int retry_cycle, time_t now) {
 			//skip per worker metrics when disabled
 			if (u_carbon.no_workers) continue;
 
-			wok = carbon_write(fd, "%s%s.%s.worker%d.requests %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, i, (unsigned long long) uwsgi.workers[i].requests, (unsigned long long) now);
+			wok = carbon_write(fd, "%s.%s%s.worker%d.requests %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, i, (unsigned long long) uwsgi.workers[i].requests, (unsigned long long) now);
 			if (!wok) goto clear;
 
 			if (uwsgi.shared->options[UWSGI_OPTION_MEMORY_DEBUG] == 1 || uwsgi.force_get_memusage) {
-				wok = carbon_write(fd, "%s%s.%s.worker%d.rss_size %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, i, (unsigned long long) uwsgi.workers[i].rss_size, (unsigned long long) now);
+				wok = carbon_write(fd, "%s.%s%s.worker%d.rss_size %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, i, (unsigned long long) uwsgi.workers[i].rss_size, (unsigned long long) now);
 				if (!wok) goto clear;
 
-				wok = carbon_write(fd, "%s%s.%s.worker%d.vsz_size %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, i, (unsigned long long) uwsgi.workers[i].vsz_size, (unsigned long long) now);
+				wok = carbon_write(fd, "%s.%s%s.worker%d.vsz_size %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, i, (unsigned long long) uwsgi.workers[i].vsz_size, (unsigned long long) now);
 				if (!wok) goto clear;
 			}
 
@@ -319,26 +319,26 @@ static void carbon_push_stats(int retry_cycle, time_t now) {
 				}
 			}
 			if (do_avg_push) {
-				wok = carbon_write(fd, "%s%s.%s.worker%d.avg_rt %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, i, (unsigned long long) avg_rt, (unsigned long long) now);
+				wok = carbon_write(fd, "%s.%s%s.worker%d.avg_rt %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, i, (unsigned long long) avg_rt, (unsigned long long) now);
 				if (!wok) goto clear;
 			}
 
-			wok = carbon_write(fd, "%s%s.%s.worker%d.tx %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, i, (unsigned long long) uwsgi.workers[i].tx, (unsigned long long) now);
+			wok = carbon_write(fd, "%s.%s%s.worker%d.tx %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, i, (unsigned long long) uwsgi.workers[i].tx, (unsigned long long) now);
 			if (!wok) goto clear;
 
-			wok = carbon_write(fd, "%s%s.%s.worker%d.busyness %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, i, (unsigned long long) worker_busyness, (unsigned long long) now);
+			wok = carbon_write(fd, "%s.%s%s.worker%d.busyness %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, i, (unsigned long long) worker_busyness, (unsigned long long) now);
 			if (!wok) goto clear;
 
-			wok = carbon_write(fd, "%s%s.%s.worker%d.harakiri %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, i, (unsigned long long) uwsgi.workers[i].harakiri_count, (unsigned long long) now);
+			wok = carbon_write(fd, "%s.%s%s.worker%d.harakiri %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, i, (unsigned long long) uwsgi.workers[i].harakiri_count, (unsigned long long) now);
 			if (!wok) goto clear;
 
 		}
 
 		if (uwsgi.shared->options[UWSGI_OPTION_MEMORY_DEBUG] == 1 || uwsgi.force_get_memusage) {
-			wok = carbon_write(fd, "%s%s.%s.rss_size %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, (unsigned long long) total_rss, (unsigned long long) now);
+			wok = carbon_write(fd, "%s.%s%s.rss_size %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, (unsigned long long) total_rss, (unsigned long long) now);
 			if (!wok) goto clear;
 
-			wok = carbon_write(fd, "%s%s.%s.vsz_size %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, (unsigned long long) total_vsz, (unsigned long long) now);
+			wok = carbon_write(fd, "%s.%s%s.vsz_size %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, (unsigned long long) total_vsz, (unsigned long long) now);
 			if (!wok) goto clear;
 		}
 
@@ -353,11 +353,11 @@ static void carbon_push_stats(int retry_cycle, time_t now) {
 			}
 		}
 		if (do_avg_push) {
-			wok = carbon_write(fd, "%s%s.%s.avg_rt %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, (unsigned long long) c_total_avg_rt, (unsigned long long) now);
+			wok = carbon_write(fd, "%s.%s%s.avg_rt %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, (unsigned long long) c_total_avg_rt, (unsigned long long) now);
 			if (!wok) goto clear;
 		}
 
-		wok = carbon_write(fd, "%s%s.%s.tx %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, (unsigned long long) total_tx, (unsigned long long) now);
+		wok = carbon_write(fd, "%s.%s%s.tx %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, (unsigned long long) total_tx, (unsigned long long) now);
 		if (!wok) goto clear;
 
 		if (active_workers > 0) {
@@ -366,18 +366,18 @@ static void carbon_push_stats(int retry_cycle, time_t now) {
 		} else {
 			total_avg_busyness = 0;
 		}
-		wok = carbon_write(fd, "%s%s.%s.busyness %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, (unsigned long long) total_avg_busyness, (unsigned long long) now);
+		wok = carbon_write(fd, "%s.%s%s.busyness %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, (unsigned long long) total_avg_busyness, (unsigned long long) now);
 		if (!wok) goto clear;
 
-		wok = carbon_write(fd, "%s%s.%s.active_workers %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, (unsigned long long) active_workers, (unsigned long long) now);
+		wok = carbon_write(fd, "%s.%s%s.active_workers %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, (unsigned long long) active_workers, (unsigned long long) now);
 		if (!wok) goto clear;
 
 		if (uwsgi.cheaper) {
-			wok = carbon_write(fd, "%s%s.%s.cheaped_workers %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, (unsigned long long) uwsgi.numproc - active_workers, (unsigned long long) now);
+			wok = carbon_write(fd, "%s.%s%s.cheaped_workers %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, (unsigned long long) uwsgi.numproc - active_workers, (unsigned long long) now);
 			if (!wok) goto clear;
 		}
 
-		wok = carbon_write(fd, "%s%s.%s.harakiri %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, (unsigned long long) total_harakiri, (unsigned long long) now);
+		wok = carbon_write(fd, "%s.%s%s.harakiri %llu %llu\n", u_carbon.hostname, u_carbon.root_node, u_carbon.id, (unsigned long long) total_harakiri, (unsigned long long) now);
 		if (!wok) goto clear;
 
 		usl->healthy = 1;
