diff --git a/plugins/fiber/uwsgiplugin.py b/plugins/fiber/uwsgiplugin.py
index ab38a37..ff0d4cb 100644
--- a/plugins/fiber/uwsgiplugin.py
+++ b/plugins/fiber/uwsgiplugin.py
@@ -24,6 +24,9 @@ else:
     CFLAGS.append('-I' + archdir)
     CFLAGS.append('-I' + archdir + '/' + arch)
     CFLAGS.append('-I' + includedir + '/' + arch)
+    archdir2 = os.popen(RUBYPATH + " -e \"require 'rbconfig';print %s::CONFIG['rubyarchhdrdir']\"" % rbconfig).read().rstrip()
+    if archdir2:
+        CFLAGS.append('-I' + archdir2)

 LDFLAGS = []
 LIBS = []
